test/functional/apps/visualize/_tsvb_time_series.ts

```diff
@@ -432,6 +432,48 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
 
         after(async () => await visualBuilder.toggleNewChartsLibraryWithDebug(false));
       });
+
+      describe('index pattern selection mode', () => {
+        it('should disable switch for Kibana index patterns mode by default', async () => {
+          await visualBuilder.clickPanelOptions('timeSeries');
+          const isEnabled = await visualBuilder.checkIndexPatternSelectionModeSwitchIsEnabled();
+          expect(isEnabled).to.be(false);
+        });
+
+        describe('metrics:useStringIndices = true', () => {
+          beforeEach(async () => {
+            await kibanaServer.uiSettings.update({ 'metrics:useStringIndices': true });
+            await browser.refresh();
+            await visualBuilder.clickPanelOptions('timeSeries');
+          });
+
+          it('should not disable switch for Kibana index patterns mode', async () => {
+            await visualBuilder.switchIndexPatternSelectionMode(true);
+
+            const isEnabled = await visualBuilder.checkIndexPatternSelectionModeSwitchIsEnabled();
+            expect(isEnabled).to.be(true);
+          });
+
+          it('should disable switch after selecting Kibana index patterns mode and metrics:useStringIndices = false', async () => {
+            await visualBuilder.switchIndexPatternSelectionMode(false);
+            await kibanaServer.uiSettings.update({ 'metrics:useStringIndices': false });
+            await browser.refresh();
+            await visualBuilder.clickPanelOptions('timeSeries');
+
+            let isEnabled = await visualBuilder.checkIndexPatternSelectionModeSwitchIsEnabled();
+            expect(isEnabled).to.be(true);
+
+            await visualBuilder.toggleIndexPatternSelectionModePopover();
+            await visualBuilder.switchIndexPatternSelectionMode(true);
+            isEnabled = await visualBuilder.checkIndexPatternSelectionModeSwitchIsEnabled();
+            expect(isEnabled).to.be(false);
+          });
+        });
+
+        after(
```

Comment:
is it required?  From my understanding `after` block should be used only if some value was changed in `before`.